shared/src/commonMain/kotlin/org/hyperskill/app/track_selection/presentation/TrackSelectionListReducer.kt

```diff
@@ -0,0 +1,60 @@
+package org.hyperskill.app.track_selection.presentation
+
+import org.hyperskill.app.track.domain.model.Track
+import org.hyperskill.app.track_selection.domain.analytic.TrackSelectionListTrackClickedHyperskillAnalyticEvent
+import org.hyperskill.app.track_selection.domain.analytic.TrackSelectionListViewedHyperskillAnalyticEvent
+import org.hyperskill.app.track_selection.presentation.TrackSelectionListFeature.Action
+import org.hyperskill.app.track_selection.presentation.TrackSelectionListFeature.InternalAction
+import org.hyperskill.app.track_selection.presentation.TrackSelectionListFeature.Message
+import org.hyperskill.app.track_selection.presentation.TrackSelectionListFeature.State
+import ru.nobird.app.presentation.redux.reducer.StateReducer
+
+internal class TrackSelectionListReducer : StateReducer<State, Message, Action> {
+    override fun reduce(state: State, message: Message): Pair<State, Set<Action>> =
+        when (message) {
+            is Message.Initialize ->
+                if (state is State.Idle ||
+                    (message.forceUpdate && (state is State.Content || state is State.NetworkError))
+                ) {
+                    State.Loading to setOf(InternalAction.Initialize)
+                } else {
+                    null
+                }
+            is TrackSelectionListFeature.TracksFetchResult.Success ->
+                if (state is State.Loading) {
+                    State.Content(message.tracks, message.selectedTrackId) to emptySet()
+                } else {
+                    null
+                }
+            TrackSelectionListFeature.TracksFetchResult.Error ->
+                if (state is State.Loading) {
+                    State.NetworkError to emptySet()
+                } else {
+                    null
+                }
+            TrackSelectionListFeature.TrackSelectionResult.Success ->
+                state to setOf(
+                    Action.ViewAction.NavigateTo.StudyPlan,
+                    Action.ViewAction.ShowTrackSelectionStatus.Success,
+                )
+            TrackSelectionListFeature.TrackSelectionResult.Error ->
+                state to setOf(Action.ViewAction.ShowTrackSelectionStatus.Error)
+            is Message.TrackClicked ->
```

Comment:
Just a reminder that we need confirmation before selecting a track https://github.com/hyperskill/mobile-app/pull/481#discussion_r1194890971